bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ReadOnlyUtils.java

```diff
@@ -233,6 +234,10 @@ private static BIntersectionType setImmutableIntersectionType(Type type, Set<Typ
                         origRecordType.flags |= SymbolFlags.READONLY, fields,
                         null, origRecordType.sealed,
                         origRecordType.typeFlags);
+                for (Map.Entry<String, BFunctionPointer<Object, ?>> field : origRecordType.getDefaultValues()
+                        .entrySet()) {
+                    immutableRecordType.setDefaultValue(field.getKey(), field.getValue());
```

Comment:
This is failing in both `2201.7.0` and master. created https://github.com/ballerina-platform/ballerina-lang/issues/41219 to address this issue.

Reply:
In addition, the `setDefaultValue` is used to store the default value function pointer which will only be called when a record value is created through runtime APIs. Currently, we use the `freezeDirect()` internal method to create immutable records. 
Therefore this will not create any issues.